File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
 int minSofar=prices[0];
        int maxProfit = 0;
        for(int i=0;i<prices.length;i++){
            minSofar = Math.min(minSofar,prices[i]);
            int profit = prices[i] - minSofar;
            maxProfit = Math.max(maxProfit,profit);
            }
        return maxProfit;
        }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxP = 0;
        int minS = prices[0];
        for(int i=0;i<prices.size();i++){
            minS = min(minS,prices[i]);
            int p = prices[i]-minS;
            maxP = max(maxP,p);
        }
        return maxP;
    }
};